charts/opentelemetry-collector/templates/service.yaml

```diff
@@ -30,4 +30,11 @@ spec:
   selector:
     {{- include "opentelemetry-collector.selectorLabels" . | nindent 4 }}
     {{- include "opentelemetry-collector.component" . | nindent 4 }}
+
+  {{- if or (eq .Values.mode "deployment") (eq .Values.mode "statefulset") }}
```

Comment:
Let's move all this logic into the helper.

Reply:
Done! :)